### PR TITLE
fixes #556, humanises connection errors a little

### LINKED ISSUE #556: Error handling message for "server is down" on kanidm
Could be a little ... neater 😄 

```
Error during authentication phase: Err(Transport(reqwest::Error { kind: Request, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("kanidm.example.com")), port: None, path: "/v1/auth", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })) }))
```

### PR DESCRIPTION
**Pull request by yaleman**
fixes #556, the `reqwest::Error` object gets really weird 😞 

- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

old error:

> Error during authentication phase: Err(Transport(reqwest::Error { kind: Request, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("kanidm.example.com")), port: None, path: "/v1/auth", query: None, fragment: None }, source: hyper::Error(Connect, ConnectError("tcp connect error", Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })) }))

new error looks like:

> [2021-08-07T12:39:25Z ERROR kanidm_ssh_authorizedkeys_direct] Failed to connect to kanidm server: error sending request for url (https://kanidm.example.com/v1/auth): error trying to connect: tcp connect error: Connection refused (os error 61)

### CODE REVIEW

File: kanidm_tools/Cargo.toml [hunk 1]
```diff
@@ -43,6 +43,8 @@ time = { version = "0.2", features = ["serde", "std"] }
 qrcode = { version = "0.12", default-features = false }
 bundy = "0.1"
 
+reqwest = { version = "0.11" }
```
Comment (Firstyear) [inline kanidm_tools/Cargo.toml:46]: Why is this needed?

Comment (yaleman) [reply kanidm_tools/Cargo.toml:46]: It's not, was used temporarily while trying to unwrap errors... removing it. :)
